### ISSUE TITLE
a DJA REST API client models.Model?

### ISSUE THREAD

**Issue by n2ygk** [django-json-api/django-rest-framework-json-api#472]
Has anyone implemented a `models.Model` that is a client of a JSON:API API along the lines of https://github.com/Yupeek/django-rest-models?  This seems to require https://github.com/AltSchool/dynamic-rest (DREST) which I'm not familiar with but looks sort of similar to DJA. Perhaps django-rest-models could be extended to support DJA as well as DREST? Maybe this is already solved?

In case you're wondering what I'm trying to accomplish, it's a multitier microservices architecture similar to what's described [here](https://medium.com/microservices-in-practice/microservices-layered-architecture-88a7fc38d3f1), [here](https://dzone.com/articles/layered-architecture-code-smells-medium) or [here](http://w3partnership.com/Blog/2016/09/09/apis-multi-tiered-application-architecture-and-microservices-an-investigation/).

In other words, the DRF database model APIs are at the bottom (persistence) layer but would be called from apps a layer above and to those apps should similarly look like a Model so all the usual ORM stuff "just works." JSON:API provides all the API HATEOAS plumbing to make this feasible.

**Comment by sliverc** [comment]
This is an older question so you might already have an answer.

I have a feeling though this should better be done on a serializer and view level and not directly on model layer. Such could use a JSON API client library like https://github.com/qvantel/jsonapi-client

Most likely out of scope for DJA at this point and might better be developed in its library.

**Comment by n2ygk** [comment]
Thanks @sliverc, I'll take a look.

**Comment by sliverc** [comment]
As this is an older question closing it. If there is a concrete idea how something like this could be integrated in DJA best to open another issue. I would prefer though for such a feature to be played around with in its own library first as outlined in https://github.com/django-json-api/django-rest-framework-json-api/issues/472#issuecomment-448222423
